library/pnotes.inc.php

```diff
@@ -26,6 +26,17 @@ function getPnoteById($id, $cols = "*")
     'order by date DESC limit 0,1', array($id));
 }
 
+/**
+ * Check a note, given its ID to see if it matches the user
+ *
+ * @param string $id the ID of the note to retrieve.
+ * @param string $user the user seeking to view the note
+ */
+function checkPnotesNoteId(int $id, string $user): bool
+{
+    return sqlQuery("SELECT `id`, `user` FROM pnotes WHERE id = ? AND user = ? AND deleted != 1", array($id, $user));
```

Comment:
to force the boolean return would do something like (the second check in the conditional is a bit of overkill but might as well :) ):
```
    $check = sqlQuery("SELECT `id` FROM pnotes WHERE id = ? AND user = ? AND deleted != 1", [$id, $user]);
    if (!empty($check['id']) && ($check['id'] == $id)) {
        return true;
    } else {
        return false;
    }
```